### DISCUSSION TITLE
Ricerca fatture per permanent_token

### DISCUSSION THREAD

**Discussion by Gabe-LS** [fattureincloud/api#63] [Q&A] [answered]
Buongiorno, sto aggiornando il collegamento tra il mio gestionale e FattureInCloud in modo da utilizzare la nuove API.

Fino ad oggi ho usato il permanent_token per creare un collegamento tra le fatture nel gestionale offline e quelle su FattureInCloud. Ora però mi sembra di capire che la ricerca delle fatture per permanent_token non è più possibile. Sbaglio? Come posso istruire il mio gestionale per trovarle se dispongo solo del permanent_token?

Grazie.

**Comment by emajo** [comment, marked as answer]
Buongiorno @Gabriele-LS, purtroppo la ricerca per *permanent_token* non è più possibile, se scrive all'assistenza clienti possiamo trovare un modo per aiutarla a convertire i suoi permanent_token in id.

**Reply by Gabe-LS** [reply]
Buongiorno, mi sono inventato un sistema in due step per trovare l'ID e, poi, eseguire le successive operazioni sul documento partendo dall'ID. Il sistema si basa su: 1) ricerca per data e numero del documento (che restituisce un solo risultato); 2) modifica del documento. Certo, sono due chiamate invece di una, ma quantomeno funziona.

**Reply by aikonfuture** [reply]
Buongiorno, anche noi nella v1 abbiamo utilizzato il permanent_token ed avevamo pensato anche alla stessa soluzione (ricerca per numero+data) con la v2. 
Abbiamo però un dubbio (non conoscendo bene l'applicazione): è possibile che per un documento possano essere variati la data o il numero (compatibilmente con la numerazione generale) ? Se la risposta è no, allora la soluzione ricerca numero+data funzionerà sempre, altrimenti potrebbe non fornire il risultato corretto.

Grazie.

**Reply by emajo** [reply]
Ciao @aikonfuture, l'unico dato di un documento che non può essere modificato è l'id, quello che consigliamo di fare è di andare a modificare tutti i documenti che avete in locale aggiungendo appunto il campo id, per fare ciò è possibile fare una ricerca per `numeration + number` (sicuramente univoci a differenza di `date + number' ), recuperare l'id e andare ad aggiungerlo, oppure è possibile contattare il nostro supporto clienti chiedendo un esportazione dei collegamenti `permanent_token -> id` che può velocizzare le vostre operazioni.

Grazie,
Emanuele

**Reply by aikonfuture** [reply]
Grazie Emanuele, se l'id non cambia mai è la giusta soluzione. In passato però c'erano situazioni in cui l'id cambiava (non ricordo adesso per quale motivo) e per questo era stato utilizzato il permanent_token, essendo l'unico dato immutabile.
